src/connector/src/source/mod.rs

```diff
@@ -42,3 +42,15 @@ pub use mock_external_table::MockExternalTableReader;
 pub use crate::source::filesystem::{S3_CONNECTOR, S3_V2_CONNECTOR};
 pub use crate::source::nexmark::NEXMARK_CONNECTOR;
 pub use crate::source::pulsar::PULSAR_CONNECTOR;
+
+pub fn is_key_belong_to_format_encode_options(key: &str) -> bool {
+    const PREFIXES: &[&str] = &[
```

Comment:
`aws.credentials.*` are not part of the options?

Reply:
`aws.credentials.*` can also exist in `with_properties`.

Reply:
But for compatibility, they may also need to be copied (back) from `with_properties` to `format_encode_options` for a legacy source. Can we test this case in some way?

Reply:
After validation by printing `SpecificParserConfig`. `format_encode_options` actually accept flatten keys like region/endpoint/.. as aws config. So we need to move these flatten keys instead of `aws.credentials.*`.
```rust
#[derive(Deserialize, Debug, Clone, WithOptions)]
pub struct AwsAuthProps {
    pub region: Option<String>,
    #[serde(alias = "endpoint_url")]
    pub endpoint: Option<String>,
    pub access_key: Option<String>,
    pub secret_key: Option<String>,
    pub session_token: Option<String>,
    pub arn: Option<String>,
    /// This field was added for kinesis. Not sure if it's useful for other connectors.
    /// Please ignore it in the documentation for now.
    pub external_id: Option<String>,
    pub profile: Option<String>,
}
```